zerver/lib/topic_mutes.py

```diff
@@ -106,34 +106,38 @@ def topic_is_muted(user_profile, stream, topic_name):
 
 def exclude_topic_mutes(conditions, user_profile, stream_name):
     # type: (List[Selectable], UserProfile, Text) -> List[Selectable]
-    muted_topics = get_topic_mutes(user_profile)
-    if not muted_topics:
-        return conditions
-
+    stream_id = None
     if stream_name is not None:
-        muted_topics = [m for m in muted_topics if m[0].lower() == stream_name]
-        if not muted_topics:
-            return conditions
+        try:
+            stream_id = get_stream(stream_name, user_profile.realm)
```

Comment:
This returns a `Stream`, not an `int`, so shouldn't be called `stream_id`.

Reply:
That should have had `.id` at the end of it.  The code fix is easy, but obviously the tests didn't catch this, so I'll add a test.